girder/utility/plugin_utilities.py

```diff
@@ -17,462 +14,94 @@
-            yield plugin
 
-
-def loadPlugin(name, root, appconf, apiRoot=None):
+class config(object):  # noqa: class name
     """
-    Loads a plugin into the application. This means allowing it to create
-    endpoints within its own web API namespace, and to register its event
-    listeners, and do anything else it might want to do.
+    Wrap a plugin's ``load`` method appending plugin metadata.
 
-    :param name: The name of the plugin (i.e. its directory name)
-    :type name: str
-    :param root: The root node of the web API.
-    :param appconf: The cherrypy configuration for the server.
-    :type appconf: dict
+    :param name str: The plugin's name
+    :param description str: A brief description of the plugin.
+    :param version str: A semver compatible version string.
+    :param dependencies list: A list of plugins required by this plugin.
+    :returns: A decorator that appends the metadata to the method
     """
-    if apiRoot is None:
-        apiRoot = root.api.v1
-
-    pluginDir = os.path.join(getPluginDir(), name)
-    isPluginDir = os.path.isdir(os.path.join(pluginDir, 'server'))
-    isPluginFile = os.path.isfile(os.path.join(pluginDir, 'server.py'))
-    pluginLoadMethod = None
+    def __init__(self, **kw):
+        self.config = kw
 
-    if not os.path.exists(pluginDir):
-        # Try to load the plugin as an entry_point
-        for entry_point in iter_entry_points(group='girder.plugin', name=name):
-            pluginLoadMethod = entry_point.load()
-            module = importlib.import_module(entry_point.module_name)
-            pluginDir = os.path.dirname(module.__file__)
-            module.PLUGIN_ROOT_DIR = pluginDir
-            girder.plugins.__dict__[name] = module
-            isPluginDir = True
+    def __call__(self, func):
+        @six.wraps(func)
+        def wrapped(*arg, **kw):
+            return func(*arg, **kw)
+        wrapped.config = self.config
+        return wrapped
 
-    if not os.path.exists(pluginDir):
-        raise Exception('Plugin directory does not exist: %s' % pluginDir)
-    if not isPluginDir and not isPluginFile:
-        # This plugin does not have any server-side python code.
-        return root, appconf, apiRoot
 
-    mailTemplatesDir = os.path.join(pluginDir, 'server', 'mail_templates')
-    if os.path.isdir(mailTemplatesDir):
-        # If the plugin has mail templates, add them to the lookup path
-        mail_utils.addTemplateDirectory(mailTemplatesDir, prepend=True)
+def getPluginConfig(plugin):
+    # TODO: Handle errors?
+    mgr = NamedExtensionManager(
+        namespace=NAMESPACE,
+        names=[plugin],
+        warn_on_missing_entrypoint=True
+    )
+    config = getattr(mgr[plugin].plugin, 'config', {})
+    defaultConfig = {
+        'name': plugin,
+        'description': '',
+        'url': '',
+        'version': '',
+        'dependencies': [],
+        'staticWebDependencies': set()
+    }
+    return dict(defaultConfig, **config)
```

Comment:
This just gets the configuration from a specific plugin using `stevedore`.  There are hooks you can add to add error handling and logging.